### DISCUSSION TITLE
GTR4 Simulator Update V1.0.4 Support Zepp OS 2.1

### DISCUSSION THREAD

**Discussion by Xuemuyang** [zepp-health/zeppos-forum#189] [Announcements]
Dear developers, our GTR4 simulator has been updated to V1.0.4 with Zepp OS 2.1. 

For new features of Zepp OS 2.1, please refer to https://docs.zepp.com/docs/guides/version-info/new-features-21/

You can simulate location related data on the simulator. Welcome to experience!

Zepp OS Team

**Comment by alexmathew11** [comment]
Link is broken. In my device it's showing error idk why

**Reply by Xuemuyang** [reply]
Thank you for your feedback, it has been fixed.

**Reply by alexmathew11** [reply]
Can you answer my questions please?

**Reply by MyoungXUE** [reply]
Thank you for sharing your code with the community. It shows your trust in us. However, the community's resources are currently limited and may not be able to fully help you thoroughly check the issues in your code. Please understand. If you have specific questions, other developers in the community should still be able to provide some opinions. 

If you can locate and simplify the issues as much as possible and provide a relatively specific problem description, other developers in the community are more likely to provide some fair advice.

Although the community's resources are limited at this time, we still want to provide you with some possible solutions or resources:

1. We added a lot number of cases in the showcase Mini Program, you can refer to https://docs.zepp.com/docs/samples/app/showcase/
2. Refer to the Best Practices section on error catching and Mini Program debugging to debug your program in conjunction with logging

https://docs.zepp.com/docs/guides/best-practice/debug/

https://docs.zepp.com/docs/guides/best-practice/error-catch/

**Reply by alexmathew11** [reply]
Hey I know but some issues cannot be solved so that's why I try to ask here and in this platform you have put an option to ask community for help that's why I try to ask here. For example check my latest question there I could not find why it sometimes shows Undefined after each response. I tried figuring out with chatgpt but still he can't solve the problem it says ask the developer, So that's thats the reason why I came here.

**Comment by XiaomaiTX** [comment]
The page cannot be visited
Is the document updated currently?
![image](https://github.com/zepp-health/zeppos-forum/assets/59403296/db449bf6-d747-4c5f-a323-2a76e9ca19de)

**Reply by Xuemuyang** [reply]
Thank you for your feedback, it has been fixed.

**Comment by joaquinperaza** [comment]
I have simulator 1.2.0 and the last gtr4 version I can download is v1.0.3 is there anything else I need to do to get the new version?

**Reply by zdunex25** [reply]
+1, they did update gtr 3 and 3 pro, but not gtr 4...

**Reply by joaquinperaza** [reply]
They shared a screenshot on slack with it. I think they forgot to release it 😅

**Reply by zdunex25** [reply]
I guess, I reported gtr mini system image to be a gtr 4 mini image instead and it took a month to fix so I was not even asking for gtr4 update :P

**Reply by joaquinperaza** [reply]
I don’t understand the point of making even this announcement to the community when it is not possible to access the update.

**Reply by XiaomaiTX** [reply]
I finished the update two days ago and saw GTR4 v1.0.4. I'm in China. Maybe the CDN in your region has not updated the cache.

**Reply by joaquinperaza** [reply]
Honestly, if that is the case, you should consider a better CDN, someday you will need an urgent rollback, and having this limitation is a significant risk

**Reply by MyoungXUE** [reply]
Sorry for the trouble caused to developers, we are working on this issue.

**Reply by MyoungXUE** [reply]
@joaquinperaza Hello, the CDN cache has been refreshed, please try again

**Reply by joaquinperaza** [reply]
Still not working, I installed the simulator from scratch, and this is all I got:
<img width="1066" alt="image" src="https://github.com/zepp-health/zeppos-forum/assets/18288382/31598928-af23-4627-84c9-b691d8bad3c9">

I even tried using a VPN to test downloading the simulator and the list of available images from a region closer to China (I'm in the US), but it did not work.

**Reply by joaquinperaza** [reply]
<img width="1051" alt="image" src="https://github.com/zepp-health/zeppos-forum/assets/18288382/433691c3-5fe4-4a7d-ad1e-364c71d05ece">

**Reply by MyoungXUE** [reply]
Sorry, we will check again.

**Reply by zdunex25** [reply]
@MyoungXUE i do no think it is related to cache because older watches has been updated right away
![image](https://github.com/zepp-health/zeppos-forum/assets/1674725/ff6eee28-9969-4897-b105-183fe196df20)

**Reply by zdunex25** [reply]
@joaquinperaza there we go, it just showed up
![image](https://github.com/zepp-health/zeppos-forum/assets/1674725/33909b61-e0d1-44bf-8f90-5937115649de)

**Reply by zdunex25** [reply]
@MyoungXUE it is still broken, i used to test my project on trex ultra and it works well on it, but on this latest gtr4 image i can only see a black screen 
![image](https://github.com/zepp-health/zeppos-forum/assets/1674725/2e34cd9f-2d5b-461d-8bcc-f43c63a93430)

**Reply by XiaomaiTX** [reply]
> @MyoungXUE it is still broken, i used to test my project on trex ultra and it works well on it, but on this latest gtr4 image i can only see a black screen 
> 
> 
> 

@zdunex25 
It seems that you have not imported the log module. This is a screenshot of the document.

![image](https://github.com/zepp-health/zeppos-forum/assets/59403296/51ac5dee-f4c0-4b6d-82b2-d9cf780afef2)

**Reply by zdunex25** [reply]
@XiaomaiTX i did, this project can work on any other system image but gtr4 1.0.4, it even starts on gtr4 1.0.3 until i initiate zeppos 2.1 features
i will try to reinstall the whole thing

**Reply by XiaomaiTX** [reply]
@zdunex25 
Can you share some of your code?

**Reply by zdunex25** [reply]
@MyoungXUE @XiaomaiTX the problem was local cache of simulator placed in %appdata%, once i cleared it everything started to work, sorry for the trouble and thanks for assistance

**Reply by XiaomaiTX** [reply]
You are welcome!👋
